lib/crypto/c_src/block.c

```diff
@@ -27,20 +27,27 @@ ERL_NIF_TERM block_crypt_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]
     struct cipher_type_t *cipherp = NULL;
     const EVP_CIPHER     *cipher;
     ErlNifBinary         key, ivec, text;
-    EVP_CIPHER_CTX*      ctx;
+    EVP_CIPHER_CTX       *ctx = NULL;
     ERL_NIF_TERM         ret;
     unsigned char        *out;
     int                  ivec_size, out_size = 0;
+    int                  cipher_len;
 
-    if (!enif_inspect_iolist_as_binary(env, argv[1], &key)
-        || !(cipherp = get_cipher_type(argv[0], key.size))
-        || !enif_inspect_iolist_as_binary(env, argv[argc - 2], &text)) {
-        return enif_make_badarg(env);
-    }
-    cipher = cipherp->cipher.p;
-    if (!cipher) {
+    if (argc != 4 && argc != 5)
+        goto bad_arg;
```

Comment:
Throwing `badarg` on invalid `argc` to NIFs are pointless.
Calling a NIF with wrong number of arguments are as doable as calling a pure Erlang functions with wrong number of arguents.
Either remove such checks or replace them with (debug) asserts crashing the VM.

Reply:
Ah, ok.  I can remove them.